packages/workflow/src/Workflow.ts

```diff
@@ -1175,17 +1175,13 @@ export class Workflow {
 		}
 
 		let connectionInputData: INodeExecutionData[] = [];
-		if (
-			nodeType.execute ||
-			nodeType.executeSingle ||
-			(!nodeType.poll && !nodeType.trigger && !nodeType.webhook)
-		) {
-			// Only stop if first input is empty for execute & executeSingle runs. For all others run anyways
+		if (nodeType.execute || (!nodeType.poll && !nodeType.trigger && !nodeType.webhook)) {
+			// Only stop if first input is empty for execute runs. For all others run anyways
 			// because then it is a trigger node. As they only pass data through and so the input-data
 			// becomes output-data it has to be possible.
 
 			if (inputData.hasOwnProperty('main') && inputData.main.length > 0) {
-				// We always use the data of main input and the first input for executeSingle
+				// We always use the data of main input and the first input for execute
```

Comment:
From a quick look, this method gets run for nodes of multiple types and the if-check seemingly applies to those that meet the two conditions of L1183, but the original comment references only `executeSingle`, which makes me think this if-check only applies to `executeSingle`? The consequent taking only the zeroth item would seem to suggest this? Not sure without testing, but if the if-check is only intended for `executeSingle`, then the check should be removed as well.

Reply:
Considering that `executeSingle` logic is still used for RoutingNode we should not remove anything that would change anything. Additionally does also the comment seem to be not up-to-date anymore. Not setting it would make it empty and break almost all expressions.